src/core/lib/security/transport/secure_endpoint.cc

```diff
@@ -171,8 +230,11 @@ static void on_read(void* user_data, grpc_error_handle error) {
   uint8_t keep_looping = 0;
   tsi_result result = TSI_OK;
   secure_endpoint* ep = static_cast<secure_endpoint*>(user_data);
+
+  ep->read_mu.Lock();
```

Comment:
the contents of these buffers need to be protected by read_mu, so this lock should cover everything consuming them (somewhere around line 302 in this version of the file)

Reply:
Done